src/Symfony/Component/Console/Application.php

```diff
@@ -179,7 +179,9 @@ public function doRun(InputInterface $input, OutputInterface $output)
 
         if (!$name) {
             $name = $this->defaultCommand;
-            $input = new ArrayInput(array('command' => $this->defaultCommand));
+            $inputArgument = $this->definition->getArgument('command');
+            $inputArgument->setMode(InputArgument::OPTIONAL);
```

Comment:
this is kind of ugly fix, that allows me to call setDefault() - it disallows setting a default value for the commands with REQUIRED value.